server/src/main/java/org/elasticsearch/health/node/HealthIndicatorDisplayStringGetter.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.health.node;
+
+import org.elasticsearch.cluster.metadata.IndexMetadata;
+import org.elasticsearch.cluster.metadata.Metadata;
+import org.elasticsearch.cluster.node.DiscoveryNode;
+
+import java.util.Comparator;
+import java.util.Locale;
+import java.util.Set;
+
+import static java.util.stream.Collectors.joining;
+
+public class HealthIndicatorDisplayStringGetter {
+
+    public static String getNodeName(DiscoveryNode node) {
+        if (node.getName() != null) {
+            return String.format(Locale.ROOT, "[%s][%s]", node.getId(), node.getName());
+        }
+        return String.format(Locale.ROOT, "[%s]", node.getId());
+    }
+
+    public static String getTruncatedIndices(Set<String> indices, Metadata clusterMetadata) {
+        final int maxIndices = 10;
+        String truncatedIndicesString = indices.stream()
+            .sorted(byPriorityThenByName(clusterMetadata))
+            .limit(maxIndices)
+            .collect(joining(", "));
+        if (maxIndices < indices.size()) {
+            truncatedIndicesString = truncatedIndicesString + ", ...";
+        }
+        return truncatedIndicesString;
+    }
+
+    /**
+     * Sorts index names by their priority first, then alphabetically by name. If the priority cannot be determined for an index then
+     * a priority of -1 is used to sort it behind other index names.
+     * @param clusterMetadata Used to look up index priority.
+     * @return Comparator instance
+     */
+    public static Comparator<String> byPriorityThenByName(Metadata clusterMetadata) {
```

Comment:
I think this method name doesn't make sense as a standalone static method (not private anymore).

Should it be `indicesComparatorByPriorityAndName` or something along those lines?

Reply:
Good idea